projects/packages/search/src/inline-search/class-inline-search-highlighter.php

```diff
@@ -72,12 +71,18 @@ public function process_results( array $results ): void {
 	/**
 	 * Filter the post title to show highlighted version.
 	 *
-	 * @param string $title The post title.
-	 * @param int    $post_id The post ID.
+	 * @param string   $title   The post title.
+	 * @param int|null $post_id Optional. The post ID. Default null.
 	 *
 	 * @return string The filtered title.
 	 */
-	public function filter_highlighted_title( string $title, int $post_id ): string {
+	public function filter_highlighted_title( string $title, ?int $post_id = null ): string {
+		// o2 is currently rendering <mark> tags in post titles, so we need to return the original.
+		$body_class = get_body_class();
+		if ( is_array( $body_class ) && in_array( 'o2', $body_class, true ) ) {
+			return $title;
+		}
+
```

Comment:
Nit: Perhaps we could simplify this because the `get_body_class` function only returns the result of `array_unique`. 
```php
		if ( in_array( 'o2', $body_class, true ) ) {
			return $title;
		}
```